scripts/scheduler_ingest.py

```diff
@@ -0,0 +1,252 @@
+# ############################################################################### #
+# Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+"""
+Classes to represent and process cycle data received from the scheduler API
+"""
+import re
+from datetime import datetime
+
+
+class TimePeriod:
+    """
+    Class to represent a generic time period
+    """
+    def __init__(self, start, end):
+        self.start = start
+        self.end = end
+
+    def __eq__(self, other):
+        return self.as_dict() == other.as_dict()
+
+    # TODO: Note to EO [#2] - I ended up not actually using     # pylint:disable=fixme
```

Comment:
I think if we are not using them lets remove them for now. They are not hard to re-implement and I don't want to bloat the code base with functionality that we don't yet require and have not immediate plan to use